src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -8541,6 +8610,69 @@ int wm_vuldet_insert_ALAS(sqlite3 *db, vu_alas_vuln *vul_it, const char *target)
     return retval;
 }
 
+int wm_vuldet_insert_deps(sqlite3 *db, dependencies *deps_it, wm_vuldet_db *parsed_oval) {
+    info_state *state_it = parsed_oval->info_states;
+    info_obj *obj_it = parsed_oval->info_objs;
+    info_test *test_it = parsed_oval->info_tests;
+    sqlite3_stmt *stmt = NULL;
+    int result = 0;
+    int retval = 0;
+
+    for (int i = 0; i < deps_it->elements; ++i) {
+        if (deps_it->test_ref[i]) {
+            char *operation = NULL; 
+            char *operation_value = NULL;
+            char *package = NULL;
+            
+            while(test_it){
+                if(strcmp(deps_it->test_ref[i], test_it->id) == 0){
+                    break;
+                }
+                test_it = test_it->prev;
+            }
+            while(state_it && test_it){
```

Comment:
`test_it` doesn't change during the loop: It can be checked only once before entering the loop.